utbot-framework/src/main/kotlin/org/utbot/engine/pc/UtSolver.kt

```diff
@@ -151,6 +177,43 @@ data class UtSolver constructor(
     //protection against solver reusage
     private var canBeCloned: Boolean = true
 
+    init {
+        val x = context.mkSymbol("x")
+        val y = context.mkSymbol("y")
+        val z = context.mkSymbol("z")
+        val xConst = context.mkConst(x, type)
+        val yConst = context.mkConst(y, type)
+        val zConst = context.mkConst(z, type)
+
+        val reflect = context.mkForall(arrayOf(xConst), isSubtypeAxiom.apply(xConst, xConst), 1, null, null, null ,null)
+        val antisymmetry = context.mkForall(arrayOf(xConst, yConst), context.mkImplies(context.mkAnd(
+            isSubtypeAxiom.apply(xConst, yConst) as BoolExpr?, isSubtypeAxiom.apply(yConst, xConst) as BoolExpr?), context.mkEq(xConst, yConst)), 1, null, null, null, null)
+        val transitivity = context.mkForall(arrayOf(xConst, yConst, zConst), context.mkImplies(context.mkAnd(
+            isSubclassAxiom.apply(xConst, yConst) as BoolExpr?, isSubclassAxiom.apply(yConst, zConst) as BoolExpr?), isSubclassAxiom.apply(xConst, zConst) as BoolExpr?), 1, null, null, null, null)
+
+        val objConst = context.mkConst("java.lang.Object", type)
+        val interfaceIsNotSubclass = context.mkForall(arrayOf(xConst, yConst), context.mkImplies(
+            context.mkAnd(context.mkNot(isInterfaceAxiom.apply(yConst) as BoolExpr?), isSubtypeAxiom.apply(xConst, yConst) as BoolExpr?, context.mkNot(context.mkEq(yConst, objConst))), // ?
+            context.mkNot(isInterfaceAxiom.apply(xConst) as BoolExpr?)
+        ),
+            1, null, null, null, null)
+        val subclassEqSubtypeForNotInterfaces = context.mkForall(arrayOf(xConst, yConst), context.mkImplies(
+            context.mkAnd(context.mkAnd(context.mkNot(isInterfaceAxiom.apply(xConst) as BoolExpr?), context.mkNot(isInterfaceAxiom.apply(yConst) as BoolExpr?))),
+            context.mkEq(isSubclassAxiom.apply(xConst, yConst) as BoolExpr?, isSubtypeAxiom.apply(xConst, yConst))
+        ),
+            1, null, null, null, null)
+
+        val notSubtypeTransitivity = context.mkForall(arrayOf(xConst, yConst, zConst), context.mkImplies(
+            context.mkNot(context.mkApp(isSubtypeAxiom, zConst, xConst) as BoolExpr),
+            context.mkOr(
+                context.mkNot(context.mkApp(isSubtypeAxiom, yConst, xConst) as BoolExpr),
+                context.mkNot(context.mkApp(isSubtypeAxiom, zConst, yConst) as BoolExpr)
+            )
+        ),
+            1, null, null, null, null)
```

Comment:
`z !isSubtype x -> (y !isSubtype x || z !isSubtype y`

Looks very odd. If it is about transitivity, where is the connection between elements? Here `x, y, z` are three random variables. I think, the right version sounds like `If there are two types `x` and `y`, where `y` is a subtype of `x`, and we have some other type `z` that is not a subtype of `x`, this `z` is not a subtype of `y` as well